docs/adr/0001-last-mile-deployment-orchestration.md

```diff
@@ -59,41 +61,79 @@ This represents a straight-forward usage of the Azure Resource Manager Deploymen
-1. **Aligned with ARM for the future** - there are plans for certain AAD function to added to ARM, this approach is likely to offer a simpler migration path to using such new features
+1. **Aligned with ARM for the future** - there are plans for (additional functionality](https://github.com/Azure/bicep/issues/3565) (outside of ARM) to be made available via the Bicep templating language. Using ARM deployment scripts is likely to offer a much simpler migration path to using such new features (i.e. swapping out a custom ARM `deploymentScript` resource for a new native one)
 1. **Option to embed within ARM template** - subject to not exceeding the overall [4MB limit](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#template-limits) for fully expanded ARM templates, this avoids additional artefact deployment considerations
+1. **Dev Inner Loop** - the script source code can be maintained as standalone scripts and only embedded into the Bicep template at the point they are used (or published to a Bicep Registry). This aligns well with current development best-practises by supporting allowing the developer to test scripts locally as well as using any existing test frameworks.
+1. **Supports re-use via Bicep modules** - encapsulating each ARM deployment script as Bicep a module that can published to a Bicep Registry, will allow the functionality to be shared and exposed via a clear interface defined by the module parameters
 
 
 #### Disadvantages:
 
 1. **Scripts may be too big to embed** - this would require the scripts to be deployed to an accessible location prior to use, which adds complexity and friction to the dev inner-loop
-1. **No defacto method for sharing deployment scripts** - a number of approaches could be adopted but no particularly developer-friendly recommendations.  The only ARM-native option appears to be the use of [TemplateSpecs](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-specs).
-1. **Currently unable to customise the container image used to execute the deployment scripts** - this means that any additional tools must be installed as part of the script, which makes them more complex, potentially more fragile and longer to run
+1. **Currently unable to customise the container image used to execute the deployment scripts** - this means that any additional tools (not available in the base images) must be installed as part of the script, which introduces more moving parts to the wrapper script itself and could make them longer to run
+1. **Difficult to re-use the scripts outside of ARM** - the scripts themselves, whilst not maintained as embedded into the Bicep template are still just a collection of 'loose' scripts that are difficult to make available to scenarios running outside of ARM
```

Comment:
I don't quite understand this: `whilst not maintained as embedded into the Bicep template`

Reply:
That's because it made no sense! :-) Now updated.